### ISSUE TITLE
Strict `void` lint request

### ISSUE THREAD

**Issue by lrhn** [dart-lang/linter#3471]

**Comment by bwilkerson** [comment]
I don't have an objection to this being implemented as a lint, but it might make more sense for it to be implemented as a language option, along the same lines as `strict-casts`, `strict-inference`, and `strict-raw-types`, possibly called `strict-void`.

@leafpetersen @srawlins

**Comment by lrhn** [comment]
Could a "language option" not just be a set of lints? I don't see any need for more categories of user-enabled/configured source analyses.

**Comment by eernstg** [comment]
Is this issue a duplicate (or a further development) of #1830?

**Comment by rakudrama** [comment]
I think this goes too far and throws out some good things with the bad.
Chaining explicit-void-to-explicit-void is a nice language feature.

Writing methods that delegate with `=>` without worrying about the 'type' is a nice affordance, allowing the delegating pattern be shorter and have visual uniformity between void and non-void methods. It is make-work, i.e. an impediment to coding velocity,  to have to flip between a method definition that has an expression body vs a full body, and having to avoid `=>` is especially ugly in closures used as arguments.

The current rules allow annotations of expression-statements by annotating a variable.
```dart
@pragma('dart2js:disable-inlining')
void _ = print('hello');
```
If the above becomes illegal then we will need another way of annotating an expression-statement.
(The pattern is not yet  in common usage but dart2js is heading towards finely scoped annotations to restrict the scope of non-spec behaviour (https://github.com/dart-lang/sdk/issues/49475). As the different annotations become more finely scoped, using this pattern would become the recommended practice.)
